src/cose/cose_sign.rs

```diff
@@ -1,37 +1,40 @@
 /// We don't need COSE signing at the moment. But we need to generate test files.
 /// This module implements basic COSE signing.
 use cose::nss;
-use cose::decoder::*;
 use cose::CoseError;
 use cose::util::{build_cose_signature, build_protected_header, build_protected_sig_header,
                  get_sig_struct_bytes};
+use cose::SignatureAlgorithm;
 
 pub fn sign(
     payload: &[u8],
-    alg: CoseSignatureType,
-    ee_cert: &[u8],
+    algorithms: &[SignatureAlgorithm],
+    ee_certs: &[&[u8]],
     cert_chain: &[&[u8]],
-    pkcs8: &[u8],
+    pkcs8: &[&[u8]],
 ) -> Result<Vec<u8>, CoseError> {
-    let nss_alg = match alg {
-        CoseSignatureType::ES256 => nss::SignatureAlgorithm::ES256,
-        CoseSignatureType::PS256 => nss::SignatureAlgorithm::PS256,
-    };
+    assert_eq!(pkcs8.len(), ee_certs.len());
```

Comment:
Looks like we should define a type that represents the trio (algorithm, certificate, key).